### PR TITLE
Wait plugin initialization with promises

### PR DESCRIPTION
**Pull request by dbengsch**
**Enhancements:**

* Wait plugin initialization with promises
* Allow all http valid methods in plugin routes

### CODE REVIEW

File: lib/api/core/plugins/pluginsManager.js [hunk 1]
```diff
@@ -98,70 +98,75 @@ function PluginsManager (kuzzle) {
...
-          debug('[%s] starting plugin in %s mode', pluginName, plugin.config.privileged ? 'privileged' : 'standard');
+    promises = Object.keys(this.plugins).map((pluginName) => {
+      const
+        plugin = this.plugins[pluginName],
+        pipeWarnTime = this.config.common.pipeWarnTime,
+        pipeTimeout = this.config.common.pipeTimeout,
+        initTimeout = this.config.common.initTimeout;
+
+      if (plugin.config.threads) {
+        debug('[%s] starting worker with %d treads', pluginName, plugin.config.threads);
```
Comment (jenow) [inline lib/api/core/plugins/pluginsManager.js]: typo on "treads"

File: lib/api/core/plugins/pluginsManager.js [hunk 2]
```diff
@@ -431,9 +439,9 @@ function initControllers (controllers, routes, plugin, pluginName) {
           return false;
         }
 
-        if (['post', 'get'].indexOf(route.verb.toLowerCase()) === -1) {
+        if (['get', 'head', 'post', 'put', 'delete', 'connect', 'options', 'trace', 'patch'].indexOf(route.verb.toLowerCase()) === -1) {
```
Comment (benoitvidis) [inline lib/api/core/plugins/pluginsManager.js]: is the `httpRouter` already able to handle all these methods? As far as I can see, only `get`, `post`, `put` and `delete` are accepted.

Further, if I am not mistaken, the `options` verb has a specific behaviour within kuzzle &/|| the proxy.

Comment (scottinet) [inline lib/api/core/plugins/pluginsManager.js]: The httpRouter now handles `head`, but you're right for the other methods.

Comment (dbengsch) [inline lib/api/core/plugins/pluginsManager.js]: @benoitvidis Done

**Pull request merged**